terraform/modules/self-service/modules/cognito/cognito.tf

```
@@ -2,9 +2,94 @@ locals {
   service = "self-service"
 }
 
+resource "aws_sns_topic" "cognito-sns-topic" {
+  name = "cognito-sns-topic"
+}
+
+resource "aws_sns_topic_policy" "cognito-sns-policy" {
+  arn = "${aws_sns_topic.cognito-sns-topic.arn}"
+
+  policy = "${data.aws_iam_policy_document.sns-topic-policy.json}"
+}
+
+data "aws_iam_policy_document" "sns-topic-policy" {
+  policy_id = "__default_policy_ID"
+
+  statement {
+    actions = [
+      "SNS:Publish"
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    resources = [
+      "${aws_sns_topic.cognito-sns-topic.arn}",
+    ]
+
+    sid = "__default_statement_ID"
+  }
+}
+
+resource "aws_iam_role" "cognito_sns_role" {
+  name               = "cognito_sns_role"
+  path               = "/service-role/"
+  assume_role_policy = <<-EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "cognito-idp.amazonaws.com"
+            },
+            "Action": [
+                "sts:AssumeRole"
+            ]
+        }
+    ]
+  }
+  EOF
+}
+
+resource "aws_iam_role_policy" "cognito_sns_role_policy" {
+  name = "cognito_sns_role_policy"
```

```suggestion
  name = "cognito-sns-role-policy"
```
sorry for being picky, I just like consistency and we tend to name the terraform resources with underscores, and the actual resource names with hyphens... see the rest of the files or other modules...